staging/src/k8s.io/apiserver/pkg/authorization/cel/compile.go

```diff
@@ -65,6 +66,20 @@ type compiler struct {
 	envSet *environment.EnvSet
 }
 
+var (
+	defaultCompilerOnce sync.Once
+	defaultCompiler     Compiler
+)
+
+// DefaultCompiler returns a shared default Compiler. Do not modify the returned Compiler.
```

Comment:
same comment about not calling this before feature gates and compatibility versions are finalized? how are we guarding against that?

Reply:
see https://github.com/kubernetes/kubernetes/pull/127504#discussion_r1774737883, then it is by construction.

Reply:
Adding a note everywhere where it depends on feature gates.